src/Core/Resolvers/Factories/MutationEngineFactory.cs

```diff
@@ -18,7 +19,15 @@ namespace Azure.DataApiBuilder.Core.Resolvers.Factories
     /// </summary>
     public class MutationEngineFactory : IMutationEngineFactory
     {
-        private readonly Dictionary<DatabaseType, IMutationEngine> _mutationEngines;
+        private Dictionary<DatabaseType, IMutationEngine> _mutationEngines;
+        private readonly RuntimeConfigProvider _runtimeConfigProvider;
+        private readonly IAbstractQueryManagerFactory _queryManagerFactory;
+        private readonly IMetadataProviderFactory _metadataProviderFactory;
+        private readonly CosmosClientProvider _cosmosClientProvider;
+        private readonly IQueryEngineFactory _queryEngineFactory;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly IAuthorizationResolver _authorizationResolver;
+        private readonly GQLFilterParser _gQLFilterParser;
```

Comment:
we have to add all these dependencies to mutationenginefactory? and all for just subscribing to the event? is there another way to accomplish what we need to do without adding a large number of dependencies in our factory classes?

Reply:
Another option that I considered was to have all of these dependencies on the Loader and then send them as arguments to the event handler.